lib/shopify_app/session/session_repository.rb

```diff
@@ -92,13 +93,46 @@ def delete_session(id)
       def load_shop_storage
         return unless @shop_storage
 
-        @shop_storage.respond_to?(:safe_constantize) ? @shop_storage.safe_constantize : @shop_storage
+        shop_storage_class =
+          @shop_storage.respond_to?(:safe_constantize) ? @shop_storage.safe_constantize : @shop_storage
+
+        [
+          :store,
+          :retrieve,
+          :retrieve_by_shopify_domain,
+        ].each do |method|
+          raise(
+            ::ShopifyApp::ConfigurationError,
+            missing_method_message("shop", method.to_s),
+          ) unless shop_storage_class.respond_to?(method)
+        end
```

Comment:
Programatically checking for required methods when loading shop_storage instead of relying on the docs to let user s know which methods are necessary for `SesssionRepository`